lib/babe/verify.go

```diff
@@ -277,12 +277,12 @@ func (b *verifier) verifyAuthorshipRight(header *types.Header) error {
 
 	preDigest, ok := preDigestItem.Value().(types.PreRuntimeDigest)
 	if !ok {
-		return fmt.Errorf("first digest item is not pre-digest")
+		return fmt.Errorf("%w: got %T", types.ErrNoFirstPreDigest, preDigestItem.Value())
```

Comment:
in some places, you've used `fmt.Errorf("%w: type=%T")` and in others you've used `fmt.Errorf("%w: got %T")` so would be nice to choose a single one and keep it as the default. IMO I would choose `fmt.Errorf("%w: got %T")` as the error explain that we did not receive the predigest but we got another type -> `ErrNoFirstPreDigest: got %T`